### ISSUE TITLE
Bug: functional tests failing with SQLite

### ISSUE THREAD

**Issue by grignards** [tracim/tracim#3064]
## Bug description & symptom

Some frontend functional tests are failing on develop

See here for details https://travis-ci.org/github/tracim/tracim/jobs/687078785

## Diagnostic

Related to a SQLite connection used in another thread:

```
message": "(sqlite3.ProgrammingError) SQLite objects created in a thread can only be used in that same thread. The object was created in thread id 139992464492288 and this is thread id 139992481277696.\n[SQL: SELECT users.password AS users_password, users.user_id AS users_user_id, users.email AS users_email, users.display_name AS users_display_name, users.created AS users_created, users.is_active AS users_is_active, users.is_deleted AS users_is_deleted, users.imported_from AS users_imported_from, users.timezone AS users_timezone, users.auth_type AS users_auth_type, users.lang AS users_lang, users.auth_token AS users_auth_token, users.auth_token_created AS users_auth_token_created, users.reset_password_token_hash AS users_reset_password_token_hash, users.reset_password_token_created AS users_reset_password_token_created, users.allowed_space AS users_allowed_space, users.profile AS users_profile \nFROM users \nWHERE users.user_id = ?]\n[parameters: [{'%(139992500389536 param)s': 1}]]\n(Background on this error at: http://sqlalche.me/e/f405)
```

## Expected

Functional tests must be green.

## Impacted components

- *backend*

**Comment by inkhey** [comment]
Can't reproduce it, either in travis or in local runned test.
Just found an unrelated (without currently any consequences) issue in travis config doing this: https://github.com/tracim/tracim/commit/6ce80ebc749f5afd759011bf29343b9f6f67785e

**Comment by Skylsmoi** [comment]
I could reproduce this issue when running cypress locally.

The test that was repeatedly reproducing it (not 100% of the time though) is: `app_file/add_file_as_user_spec.js` [here](https://github.com/tracim/tracim/blob/develop/functionnal_tests/cypress/integration/app_file/add_file_as_user_spec.js)

adding `email.processing_mode = async` to cypress_test.ini resolved the issue

**Comment by inkhey** [comment]
to reproduce the issue waitress threads should not be set to 1 :
```
[server:main]
use = egg:waitress#main
listen = %(basic_setup.listen)s
threads = 1  #this line need to be removed to use default 4 threads
```

**Comment by PhilippeAccorsi** [comment]
> adding `email.processing_mode = async` to cypress_test.ini resolved the issue

If its solved your issue, its because with "sync", backend send email and wait its finished to continue working for other requests.

**Comment by Skylsmoi** [comment]
to run this cypress test:
- check in configEnv.js that apiUrl is: `/api/v2` (a relative adress)
- (just to be sure) rebuild the frontend: `cd frontend && yarn run build`
- check that `cypress_test.ini` has the property `basic_setup.website_base_url =  http://localhost:7999`
- run the backend with cypress config: `cd backend && pserve cypress_test.ini`
- run pushpin server with docker (see @inkhey for details)
- start cypress: `cd functionnal_tests && yarn run cypress-open`
- click on the test `app_file/add_file_as_user_spec.js`

**Comment by grignards** [comment]
This bug is likely due to temporary code for TLMs publishing that keeps a reference to a sqlalchemy's session. Will be handled by #3100.

**Comment by inkhey** [comment]
@grignards can we consider this bug as fixed now as #3125  is merged ?

**Comment by grignards** [comment]
> @grignards can we consider this bug as fixed now as #3125 is merged ?

I think yes to ensure it we can re-enable threads in waitress during cypress tests (I planned to do that in this issue).
